bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBridgeHandler.java

```diff
@@ -127,14 +125,18 @@ public synchronized void login() {
             return;
         }
 
+        if (tooManyRequests || ThingStatus.ONLINE.equals(thing.getStatus())) {
+            return;
+        }
+
         try {
             if (httpClient.isStarted()) {
                 httpClient.stop();
             }
             httpClient.start();
 
             url = TAHOMA_URL + "login";
-            String urlParameters = "userId=" + thingConfig.getEmail() + "&userPassword=" + thingConfig.getPassword();
+            String urlParameters = "userId=" + thingConfig.getEmailUrlEncoded() + "&userPassword=" + thingConfig.getPasswordUrlEncoded();
```

Comment:
Maybe move the URL encoding logic to this method since it's only used here? That way the configuration object can stay a POJO.

Reply:
moved